main.js

```diff
@@ -0,0 +1,251 @@
+$(document).ready(function() {
+
+  var currentCityWeatherData = [];
+  var fiveDayCityWeatherData = [];
+
+  $('.search').on('click', function() {
+    $('.processing').css('display', 'block');
+    var $currentCity = $('#search-query').val();
+    
+    currentCityWeatherData = [];
+    fiveDayCityWeatherData = [];
+
+    $('#search-query').val('');
+    $('.set-default').css('display', 'block');
+
+    fetchDailyWeather($currentCity);
+    fetchFiveDayForecast($currentCity);
+  });
+
+  $('.find-me').on('click', function() {
+    $('.processing').css('display', 'block');
+
+    currentCityWeatherData = [];
+    fiveDayCityWeatherData = [];
+    
+    $('.set-default').css('display', 'block');
+
+    if(navigator.geolocation) {
+        navigator.geolocation.getCurrentPosition(function(position) {
+          var latitude = position.coords.latitude;
+          var longitude = position.coords.longitude;
+
+          fetchByLocation(latitude, longitude);
+          fetchFiveDayForecastByLocation(latitude, longitude);
+        });
+    } else {
+        alert("Sorry, your location was not found.");
+    }
+  });
+
+  $('#search-query').keypress(function(event) {
+    if (event.keyCode === 13) {
+        $('.search').click();
+    }
+  });
+
+  $('.set-default').on('click', function () {
+      var defaultCity = currentCityWeatherData[0].city;
+      localStorage.setItem('cityName', defaultCity);
+      alert(defaultCity + ' is now your default city.');
+  }); 
+
+  window.onload = function() {
+    var defaultCity = localStorage.getItem('cityName');
+    if (defaultCity !== null) {
+      $('#search-query').val(defaultCity); 
+      $('.search').click();
+      $('.set-default').css('display', 'none');
+    }
+  }
+
+  $(document).ajaxComplete(function () {
+    $('.processing').css('display', 'none');
+  });
+
+  var addCurrentCityWeatherData = function (data) {
+    var currentWeatherImageCode = data.weather[0].icon;
+    var imageSrc = 'http://openweathermap.org/img/wn/' + currentWeatherImageCode +'@2x.png';
+    $('#current-weather-icon').attr('src', imageSrc);
+    var weather_condition = data.weather[0].main;
+
+    var city = data.name;
+    var mapSrc = 'https://www.google.com/maps/embed/v1/place?key=AIzaSyCDyFxxpN5RtNUPm8u2aCKZ8XMqc9mfxdE&q=' + city;
```

Comment:
For future projects, you want to avoid including api keys directly in your code for security purposes. Not something you should worry about right now, but just making you aware. This will be talked about more when you get into node.js.